services/__tests__/country.spec.js

```diff
@@ -0,0 +1,171 @@
+import { jest } from '@jest/globals';
+
+import { dataRequest } from 'utils/request';
+
+import {
+  getCountriesProvider,
+  getRegionsProvider,
+  getSubRegionsProvider,
+  getFAOCountriesProvider,
+  getCountryLinksProvider,
+  getCategorisedCountries,
+  getCountryLinksSerialized,
+  GADM_DATASET,
+  SQL_QUERIES,
+} from 'services/country';
+import countryLinks from 'services/country-links.json';
+
+jest.mock('utils/request', () => ({
+  dataRequest: {
+    get: jest.fn(),
+  },
+}));
+
+jest.mock('services/country-links.json', () => ({
+  rows: [
+    {
+      iso: 'CMR',
+      external_links:
+        '[{  "title": "Interactive Forest Atlas of Cameroon",  "url": "http://cmr.forest-atlas.org/"}]',
+    },
+    {
+      iso: 'CAF',
+      external_links:
+        '[{ "title": "Interactive Forest Atlas of Central African Republic",  "url": "http://caf.forest-atlas.org/"}]',
+    },
+  ],
+}));
+
+describe('country service', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('getCountriesProvider', () => {
+    it('should fetch countries data', async () => {
+      const mockResponse = { data: [{ iso: 'BRA', name: 'Brazil' }] };
+      dataRequest.get.mockResolvedValue(mockResponse);
+
+      const result = await getCountriesProvider();
+
+      expect(dataRequest.get).toHaveBeenCalledWith(
+        `${GADM_DATASET}?sql=${SQL_QUERIES.getGADMCountries}`
+      );
+      expect(result).toEqual(mockResponse);
+    });
+  });
+
+  describe('getRegionsProvider', () => {
+    it('should fetch regions data based on adm0', async () => {
+      const mockResponse = { data: [{ id: 'BRA.25', name: 'São Paulo' }] };
+      dataRequest.get.mockResolvedValue(mockResponse);
+
+      const result = await getRegionsProvider({ adm0: 'BRA' });
+
+      expect(dataRequest.get).toHaveBeenCalledWith(
+        `${GADM_DATASET}?sql=${SQL_QUERIES.getGADMRegions}`.replace(
```

Comment:
I don't really like the `replace` calls.

The reason is that it's a duplication of how the `getRegionsProvider` method is implemented internally. This test could break if we decide to change how the values get replaced someday even though the interface with `dataRequest` remains unchanged.

I would recommend always using literal values in `expect` matchers:
`SELECT name_1 AS name, gid_1 AS id FROM gadm_administrative_boundaries WHERE adm_level='1' AND gid_0 = 'BRA' ORDER BY name`